monitor/consul/watcher.go

```diff
@@ -67,47 +67,68 @@ func New(addr, dc, token string, timeout time.Duration, ii ...Item) (*Watcher, e
 }
 
 // Watch key and prefixes for changes.
-func (w *Watcher) Watch(ctx context.Context, ch chan<- []*change.Change) error {
+func (w *Watcher) Watch(ctx context.Context) (<-chan []change.Change, error) {
 	if ctx == nil {
-		return errors.New("context is nil")
+		return nil, errors.New("context is nil")
 	}
-	if ch == nil {
-		return errors.New("change channel is nil")
-	}
-	for _, i := range w.ii {
-		var pl *watch.Plan
-		var err error
-		switch i.tp {
```

Comment:
during development I found a bug here:

If the item type was not filled with one of the two valid types a `nil` plan would be inserted in the slice only to later issue a segfault. I have corrected the behavior with proper error handling (early return if type does not comply with the two supported types). You can find also a new unit-test covering this